<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ GenomicOriginsScripts</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />
<meta property="og:image" content="https://k-hench.github.io/GenomicOriginsScripts/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicOriginsScripts</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.90038</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/k-hench/GenomicOriginsScripts/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-project-configuration" class="hasAnchor"><a href="#section-project-configuration" class="anchor"></a>Project configuration</h2>
          <p class="section-desc"><p>General settings that are used to keep plots consistent.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="clr.html">clr</a></code> </p>
        </td>
        <td><p>Hamlet colors (standard)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clr2.html">clr2</a></code> </p>
        </td>
        <td><p>Hamlet colors (alternative)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clr_below.html">clr_below</a></code> </p>
        </td>
        <td><p>Default background bar color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clr_loc.html">clr_loc</a></code> </p>
        </td>
        <td><p>Location colors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fll_fun.html">fll_fun()</a></code> </p>
        </td>
        <td><p>Fill function for adxiture classes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fll_n.html">fll_n()</a></code> </p>
        </td>
        <td><p>Named fill function for adxiture classes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loc_names.html">loc_names</a></code> </p>
        </td>
        <td><p>Location names</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="outlr_clr.html">outlr_clr</a></code> </p>
        </td>
        <td><p>Outlier color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_clr.html">plot_clr</a></code> </p>
        </td>
        <td><p>Default plot color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_size.html">plot_size</a></code> </p>
        </td>
        <td><p>Default plot size</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pop_levels.html">pop_levels</a></code> </p>
        </td>
        <td><p>Population order</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="project_case.html">project_case()</a></code> </p>
        </td>
        <td><p>Project case</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="project_inv_case.html">project_inv_case()</a></code> </p>
        </td>
        <td><p>Inverted project case</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="shps.html">shps</a></code> </p>
        </td>
        <td><p>Location shapes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sp_labs.html">sp_labs</a></code> </p>
        </td>
        <td><p>Species labels</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sp_names.html">sp_names</a></code> </p>
        </td>
        <td><p>Species names</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-general-functions" class="hasAnchor"><a href="#section-general-functions" class="anchor"></a>General functions</h2>
          <p class="section-desc"><p>Functions that are used at differnet places.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="add_gpos.html">add_gpos()</a></code> </p>
        </td>
        <td><p>Add genomic position</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="darken.html">darken()</a></code> </p>
        </td>
        <td><p>Darken a given color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_dxy.html">get_dxy()</a></code> </p>
        </td>
        <td><p>Import dxy data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_gxp.html">get_gxp()</a></code> </p>
        </td>
        <td><p>Import genotype x phenotype association</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_fst.html">get_fst()</a></code> </p>
        </td>
        <td><p>Import Fst data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_fst_summary_data.html">get_fst_summary_data()</a></code> </p>
        </td>
        <td><p>Import fst summary data for outlier windows</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_pi.html">get_pi()</a></code> </p>
        </td>
        <td><p>Import diversity data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="join_list.html">join_list()</a></code> </p>
        </td>
        <td><p>Iterate left_join over list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lighten.html">lighten()</a></code> </p>
        </td>
        <td><p>Lighten a given color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="process_input.html">process_input()</a></code> </p>
        </td>
        <td><p>Process script input</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="refactor.html">refactor()</a></code> </p>
        </td>
        <td><p>Reorder factor levels</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-tables" class="hasAnchor"><a href="#section-tables" class="anchor"></a>Tables</h2>
          <p class="section-desc"><p>Functions needed to compile tables.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="as_multirow.html">as_multirow()</a></code> </p>
        </td>
        <td><p>Wrap content in multirow</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="export_2_latex.html">export_2_latex()</a></code> </p>
        </td>
        <td><p>Export latex tables</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_genes.html">get_genes()</a></code> </p>
        </td>
        <td><p>Collect hamlet genes</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-figure-" class="hasAnchor"><a href="#section-figure-" class="anchor"></a>Figure 1</h2>
          <p class="section-desc"><p>Functions specific to Figure 1.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="summarize_dxy.html">summarize_dxy()</a></code> </p>
        </td>
        <td><p>Import and summarize dxy data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summarize_fst.html">summarize_fst()</a></code> </p>
        </td>
        <td><p>Import and summarize Fst data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_faint_clr.html">make_faint_clr()</a></code> </p>
        </td>
        <td><p>make a faint version of a color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_fish.html">plot_fish()</a></code> </p>
        </td>
        <td><p>Plot single fish</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_network.html">plot_network()</a></code> </p>
        </td>
        <td><p>Plot pariwise species comparisons per location</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="whisker_points.html">whisker_points()</a></code> </p>
        </td>
        <td><p>Whisker points</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-figure-" class="hasAnchor"><a href="#section-figure-" class="anchor"></a>Figure 2</h2>
          <p class="section-desc"><p>Functions specific to Figure 2.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_twisst_data.html">get_twisst_data()</a></code> </p>
        </td>
        <td><p>Import a single topology weighting data set</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="match_twisst_files.html">match_twisst_files()</a></code> </p>
        </td>
        <td><p>Import all topology weighting data sets for a location</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-figure-" class="hasAnchor"><a href="#section-figure-" class="anchor"></a>Figure 3</h2>
          <p class="section-desc"><p>Functions specific to Figure 3.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="ax_scl.html">ax_scl()</a></code> </p>
        </td>
        <td><p>Scale bp to Mb</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="custom_annoplot.html">custom_annoplot()</a></code> </p>
        </td>
        <td><p>Custom annotation base plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cross_spec.html">cross_spec()</a></code> </p>
        </td>
        <td><p>Cross species pairs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="distances_tree.html">distances_tree()</a></code> </p>
        </td>
        <td><p>Get distance matrix of a topology</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dist_tibble.html">dist_tibble()</a></code> </p>
        </td>
        <td><p>Create tibble of pairwise species distances</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_clr.html">get_clr()</a></code> </p>
        </td>
        <td><p>Defin topology weighing base color</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_dist.html">get_dist()</a></code> </p>
        </td>
        <td><p>Create distance of species pair</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_gxp_long.html">get_gxp_long()</a></code> </p>
        </td>
        <td><p>Import genotype x phenotype association in long format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_isolated_topos.html">get_isolated_topos()</a></code> </p>
        </td>
        <td><p>get isolated topologies</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_neighbour_topos.html">get_neighbour_topos()</a></code> </p>
        </td>
        <td><p>Get pairwise highlighing topologies</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is_isolated.html">is_isolated()</a></code> </p>
        </td>
        <td><p>Determine if species is isolated in topology</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="min_dist.html">min_dist()</a></code> </p>
        </td>
        <td><p>Find minimum distance to other species in topology</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="no_title.html">no_title()</a></code> </p>
        </td>
        <td><p>drop ggplot y axis titles</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_fish_zoom.html">plot_fish_zoom()</a></code> </p>
        </td>
        <td><p>Plot fish for topology legend</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_curtain.html">plot_curtain()</a></code> </p>
        </td>
        <td><p>Manage plot panels for a single outlier window</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_leg.html">plot_leg()</a></code> </p>
        </td>
        <td><p>Create topology legend</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_panel_anno.html">plot_panel_anno()</a></code> </p>
        </td>
        <td><p>Create annotation panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_panel_delta_dxy.html">plot_panel_delta_dxy()</a></code> </p>
        </td>
        <td><p>Create the delta dxy panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_panel_dxy.html">plot_panel_dxy()</a></code> </p>
        </td>
        <td><p>Create the dxy panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_panel_fst.html">plot_panel_fst()</a></code> </p>
        </td>
        <td><p>Create the fst panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_fst_poptree.html">plot_fst_poptree()</a></code> </p>
        </td>
        <td><p>Create the fst population tree panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_panel_gxp.html">plot_panel_gxp()</a></code> </p>
        </td>
        <td><p>Create the genotype x phenotype panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_panel_twisst.html">plot_panel_twisst()</a></code> </p>
        </td>
        <td><p>Create the topology weighting panel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prep_data.html">prep_data()</a></code> </p>
        </td>
        <td><p>Prepate topology weighting data for zoom plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="theme_panels.html">theme_panels()</a></code> </p>
        </td>
        <td><p>ggplot layout template</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-figure-" class="hasAnchor"><a href="#section-figure-" class="anchor"></a>Figure 4</h2>
          <p class="section-desc"><p>Functions specific to Figure 4.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_cc.html">get_cc()</a></code> </p>
        </td>
        <td><p>Import cross coalescence data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_msmc.html">get_msmc()</a></code> </p>
        </td>
        <td><p>Import msmc data</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X1</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X1.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="geom_hypo_grob2.html">geom_hypo_grob2()</a></code> </p>
        </td>
        <td><p>The custom grob geom</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_fishes_location.html">plot_fishes_location()</a></code> </p>
        </td>
        <td><p>Plot two fishes with location</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summarise_model.html">summarise_model()</a></code> </p>
        </td>
        <td><p>summarise model statistics</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X2</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X2.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="read_sc.html">read_sc()</a></code> </p>
        </td>
        <td><p>import sc data sheet</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X3</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X3.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="anno_pair_flag.html">anno_pair_flag()</a></code> </p>
        </td>
        <td><p>create hamlet pair annotations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fst_bar_row_run.html">fst_bar_row_run()</a></code> </p>
        </td>
        <td><p>crate global fst bar table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_pair_run.html">plot_pair_run()</a></code> </p>
        </td>
        <td><p>create hamlet pair annotations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="refactor_run.html">refactor_run()</a></code> </p>
        </td>
        <td><p>reorder species pairs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rescale_fst.html">rescale_fst()</a></code> </p>
        </td>
        <td><p>normalize global fst</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X4</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X4.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="fish_plot.html">fish_plot()</a></code> </p>
        </td>
        <td><p>hamlet pair annotation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fish_plot2.html">fish_plot2()</a></code> </p>
        </td>
        <td><p>hamlet pair annotation 2</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X7</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X7.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="root_manual.html">root_manual()</a></code> </p>
        </td>
        <td><p>reroot tree manually</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="setup_tree_data.html">setup_tree_data()</a></code> </p>
        </td>
        <td><p>setup phylo tree data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_tree.html">get_tree()</a></code> </p>
        </td>
        <td><p>import phylogenetic tree</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="StatTreeData.html">StatTreeData</a></code> </p>
        </td>
        <td><p>ggplot layer for node support</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="geom_nodepoint_support.html">geom_nodepoint_support()</a></code> </p>
        </td>
        <td><p>backbone for ggplot layer for node support</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X8</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X8.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="collapse_peaks.html">collapse_peaks()</a></code> </p>
        </td>
        <td><p>Collapse all Fst peaks of a given run</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_fst_fixed.html">get_fst_fixed()</a></code> </p>
        </td>
        <td><p>Compute summary stats an absolute for Fst outlier</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="reformat_run_name.html">reformat_run_name()</a></code> </p>
        </td>
        <td><p>Reformat run name</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X9</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X9.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="add_spec_drawing.html">add_spec_drawing()</a></code> </p>
        </td>
        <td><p>Add fish annotation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="adm_plot.html">adm_plot()</a></code> </p>
        </td>
        <td><p>Plot adimxture data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="data_amdx.html">data_amdx()</a></code> </p>
        </td>
        <td><p>Get adimxture data</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X10</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X10.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="getPofZ.html">getPofZ()</a></code> </p>
        </td>
        <td><p>Get hybridization data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_loc.html">plot_loc()</a></code> </p>
        </td>
        <td><p>Plot hybridization data</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-suppl-figure-x-" class="hasAnchor"><a href="#section-suppl-figure-x-" class="anchor"></a>Suppl. Figure X11</h2>
          <p class="section-desc"><p>Functions specific to Suppl. Figure X11.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_gxp_both_models.html">get_gxp_both_models()</a></code> </p>
        </td>
        <td><p>Get gxp data</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://k-hench.github.io/kh/'>Kosmas Hench</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


